scraper/main.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+from scrapy.crawler import CrawlerRunner
+from scraper.spiders.meps import XMLParser, MEPSCrawler
+from scrapy.utils.project import get_project_settings
+from scrapy.settings import Settings
+from twisted.internet import reactor, defer
+from scrapy.utils.log import configure_logging
+from scrapy import signals
+
+import argparse
+import logging
+import csv
+import os
+
+BASE_URL = 'https://www.europarl.europa.eu/meps/en/'
+count_declarations = 0
+count_meps = 0
+
+
+def update_stats(spider):
+    global count_meps
+    count_meps = count_meps + 1
+
+    if spider.has_declaration:
+        global count_declarations
+        count_declarations = count_declarations + 1
+
+
+@defer.inlineCallbacks
+def crawl(runner, data_dir):
+    yield runner.crawl(XMLParser)
+
+    with open(os.path.join(data_dir, 'meps.csv')) as f:
+        csv_data = csv.reader(f)
+        for line in csv_data:
+            full_name, id = line
+            dir_name = f'{data_dir}/{full_name} - {id}'
+            url = BASE_URL + id
+
+            crawler_object = runner.create_crawler(MEPSCrawler)
+            crawler_object.signals.connect(update_stats,
+                                           signals.spider_closed)
+
+            yield runner.crawl(crawler_object, url=url, dir_name=dir_name)
+
+    reactor.stop()
+
+
+@defer.inlineCallbacks
+def crawl_id(runner, data_dir, id):
+    crawler_object = runner.create_crawler(MEPSCrawler)
+    crawler_object.signals.connect(update_stats, signals.spider_closed)
+
+    dir_name = f'{data_dir}/TEST - {id}'
+    url = BASE_URL + id
+
+    yield runner.crawl(crawler_object, url=url, dir_name=dir_name)
+
+    reactor.stop()
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--id', nargs=1, help='MEP id')
+    args = parser.parse_args()
+
+    os.environ['SCRAPY_SETTINGS_MODULE'] = 'scraper.settings'
+    project_settings = Settings(get_project_settings())
+
+    configure_logging(project_settings)
+    logging.getLogger('scrapy').setLevel(logging.WARNING)
```

Comment:
I changed it here because scrapy shows a lot of INFO logs of its own, making mine disappear in the clutter. So I wanted them to go away and only show serious problems.

Reply:
If I set it in `settings.py` to `WARNING` it will disable my `INFO` logs too. If I remove this last line from here I will get a lot of `INFO` logs from scrapy's internal structure which will obfuscate mine. I don't really know any workaround.

Reply:
Is just leaving the spam there an option? It might be fine. Yours won't be obfuscated since they're marked with `main` or whatever your module name is.